src/neptune_scale/api/attribute.py

```diff
@@ -0,0 +1,243 @@
+import functools
+import itertools
+import warnings
+from datetime import datetime
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Callable,
+    Collection,
+    Dict,
+    Iterator,
+    List,
+    Optional,
+    Set,
+    Tuple,
+    Union,
+    cast,
+)
+
+if TYPE_CHECKING:
+    from neptune_scale.api.run import Run
+
+
+def warn_unsupported_params(fn: Callable) -> Callable:
+    # Perform some simple heuristics to detect if a method is called with parameters
+    # that are not supported by Scale
+    warn = functools.partial(warnings.warn, stacklevel=3)
+
+    def wrapper(*args, **kwargs):  # type: ignore
+        if kwargs.get("wait") is not None:
+            warn("The `wait` parameter is not yet implemented and will be ignored.")
+
+        extra_kwargs = set(kwargs.keys()) - {"wait", "step", "timestamp", "steps", "timestamps"}
+        if extra_kwargs:
+            warn(
+                f"`{fn.__name__}()` was called with additional keyword argument(s): `{', '.join(extra_kwargs)}`. "
+                "These arguments are not supported by Neptune Scale and will be ignored."
+            )
+
+        return fn(*args, **kwargs)
+
+    return wrapper
+
+
+# TODO: proper typehinting
+ValueType = Any  # Union[float, int, str, bool, datetime, Tuple, List, Dict, Set]
+
+
+class AttributeStore:
+    def __init__(self, run: "Run") -> None:
+        self._run = run
+        self._attributes: Dict[str, Attribute] = {}
+
+    def __getitem__(self, key: str) -> "Attribute":
+        path = cleanup_path(key)
+        attr = self._attributes.get(key)
```

Comment:
```
        attr = self._attributes.get(path)
```

Reply:
This is already refactored in another branch as well :-)